Improve ram estimation for insert-on-conflict-update (#17754)
crate/crate: docs/appendices/release-notes/5.10.5.rst

Fixes
=====

> - Fixed ram estimation for insert-on-conflict-update statements by using table
>   statistics. This should prevented nodes from running out of memory when running
>   large inserts with on-conflict-update clauses.
> 

Reviewer: ```suggestion
- Fixed memory estimation for ``INSERT INTO`` statements with ``ON CONFLICT``
  clause. This should help prevent nodes from running out of memory.

```
